README.md

```diff
@@ -29,40 +30,41 @@ Installation of the [npm package](https://npmjs.org/package/reaclette):
 ```js
 import React from "react";
 import { render } from "react-dom";
-import { injectState, provideState } from "reaclette";
-
-const wrapComponentWithState = provideState({
-  initialState: () => ({ counter: 0 }),
-  effects: {
-    addOne: () => (state, props) => ({ counter: state.counter + 1 }),
-  },
-  computed: {
-    square: (state, props) => state.counter * state.counter,
+import { withStore } from "reaclette";
+
+const Component = withStore(
+  {
+    initialState: props => ({ counter: 0 }),
+    effects: {
+      addOne() {
+        this.state.counter += 1;
+      },
+    },
+    computed: {
+      square: (state, props) => state.counter * state.counter,
+    },
   },
-});
-
-const Parent = wrapComponentWithState(() => <Child />);
-
-const Child = injectState(({ effects, state }) => (
-  <div>
-    <p>Our counter is at: {state.counter}</p>
-    <p>Its squared value is: {state.square}</p>
-    <p>
-      <button onClick={effects.addOne}>Add one</button>
-    </p>
-  </div>
-));
+  ({ effects, state }, props) => (
+    <div>
+      <p>Our counter is at: {state.counter}</p>
+      <p>Its squared value is: {state.square}</p>
+      <p>
+        <button onClick={effects.addOne}>Add one</button>
+      </p>
+    </div>
+  )
+);
 
-render(<Parent />, document.body);
+render(<Component />, document.body);
 ```
 
 ## API
 
 ```js
-import { provideState, injectState } from "reaclette";
+import { withStore } from "reaclette";
 ```
 
-### `provideState(options) => (Component => Component)`
+### `withStore(options, ({ state, effects }, props) => Component)`
```

Comment:
Let's be pedantic: the render function does not return a component but a React node.